sphinx/writers/latex.py

```diff
@@ -1444,7 +1445,9 @@ def visit_entry(self, node):
             self.needs_linetrimming = 1
         if len(node.traverse(nodes.paragraph)) >= 2:
             self.table.has_oldproblematic = True
-        if isinstance(node.parent.parent, nodes.thead):
+        if isinstance(node.parent.parent, nodes.thead) or \
+           node.parent.parent.parent.stubs[cell.col]:
+            # "stubs" list is an attribute of the tgroup element
             if len(node) == 1 and isinstance(node[0], nodes.paragraph) and node.astext() == '':
                 pass
             else:
```

Comment:
We could do

    self.body.append('\\sphinxstylethead{\\sphinxstyletheadfamily\n')

and in sphinx.sty

    \let\sphinxstylethead\empty
    \newcommand*\sphinxstyletheadfamily{\ttfamily}

This way backwards compatibility is maintained (the extra `\ttfamily` does no harm) and default behaviour is compatible with multiple paragraphs in table head cells.

I think I answered at least to one stackoverflow question about colorizing header cells, and this used `\sphinxstylethead`, so removing it would indeed break at least some documents. Later or even at 1.6, we can possibly deprecate `\sphinxstylethead`, and tell people to use `\sphinxstyletheadfamily` instead. If so, then removal at 1.7 ?